src/lib/BUTTON/devButton.cpp

```diff
@@ -100,6 +65,54 @@ static int start()
     {
         return DURATION_NEVER;
     }
+
+    if (GPIO_PIN_BUTTON != UNDEF_PIN)
+    {
+        button1.init(GPIO_PIN_BUTTON, GPIO_BUTTON_INVERTED);
+        button1.OnShortPress = [](){ handlePress(0, false, button1.getCount()); };
+        button1.OnLongPress = [](){ handlePress(0, true, button1.getCount()); };
+#if defined(TARGET_TX)
+        const tx_button_color_t *button_actions = config.GetButtonActions(0);
+        if (button_actions->val.actions[0].action == ACTION_NONE && button_actions->val.actions[1].action == ACTION_NONE)
+        {
+            // Set defaults for button 1
+            tx_button_color_t default_actions = {
+                .val = {
+                    .color = 0,
+                    .actions = {
+                        {false, 2, ACTION_BIND},
+                        {true, 0, ACTION_INCREASE_POWER}
+                    }
+                }
+            };
+            config.SetButtonActions(0, &default_actions);
+        }
+#endif
+    }
+    if (GPIO_PIN_BUTTON2 != UNDEF_PIN)
+    {
+        button2.init(GPIO_PIN_BUTTON2, GPIO_BUTTON_INVERTED);
+        button2.OnShortPress = [](){ handlePress(1, false, button2.getCount()); };
+        button2.OnLongPress = [](){ handlePress(1, true, button2.getCount()); };
+#if defined(TARGET_TX)
+        const tx_button_color_t *button_actions = config.GetButtonActions(1);
+        if (button_actions->val.actions[0].action == ACTION_NONE && button_actions->val.actions[1].action == ACTION_NONE)
```

Comment:
I see what you're doing here, but we shouldn't have config.SetDefaults() spread all over our codebase. Just move the defaults into config.SetDefaults(). There is no need to increment the config version for this, because said hardware will either ship with this functioning already (and thus have the proper default built in), or will ship with no default if we don't get this out in time, which is easy enough for a user to change.

This code also will keep bringing this default back if the user hates the buttons and wants no action for them.

Reply:
The problem that we have then is that there would be a circular dependency. config would need to include button to be able to put in the default values., but button depends on config so it can get the configuration.
PIO shits itself, and this has been the cause of so many problems for us.

Reply:
I moved the actions enum into `common.h` to avoid the circular dependency, and the default config is now in `config.cpp`, where it belongs.
Also updated the wifi code so you can pass /reset?config to reset the config to defaults as well.